src/js/media-controller.js

```diff
@@ -496,8 +499,29 @@ const getMediaUIAttributesFrom = (child) => {
   );
 };
 
+const hasMediaUIProps = (mediaStateReceiverCandidate) => {
+  if (
+    mediaStateReceiverCandidate.nodeName?.includes('-')
+    && !!window.customElements.get(mediaStateReceiverCandidate.nodeName?.toLowerCase())
+    && !(mediaStateReceiverCandidate instanceof window.customElements.get(mediaStateReceiverCandidate.nodeName.toLowerCase()))
+  ) {
+    window.customElements.upgrade(mediaStateReceiverCandidate);
```

Comment:
my s1 here is that we shouldn't upgrade every media state receiver to accept state.

this is a solution for that 
https://github.com/muxinc/media-chrome/blob/ef110153af0d0d6f8fb5b9fb11b4e8e0a3a9607d/src/js/media-theme-element.js#L103-L111

or keep using attributes for propagating state  😅 

---

just checking the code base, we do another unneeded upgrade here 
https://github.com/muxinc/media-chrome/blob/7930c204943985fd568fde377243ea1949456ec8/src/js/media-controller.js#L485-L485
we can get that with`customElements.get('m-c').observedAttributes`

Reply:
Per out of band convo, we're planning on at least discussing these orthogonal architectural changes before merging this PR.